qiita_db/base.py

```diff
@@ -128,28 +130,36 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
-
-    def __init__(self, id_):
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        # The value that we want is the result of the last SQL query,
+        # and it is stored in the first value of the first row
+        return trans.execute()[-1][0][0]
```

Comment:
@wasade @antgonza @ElDeveloper what do you think about this solution?
`trans.execute()` will return a triple nested list while
`trans.execute(return_one=True)` will return a single value (the same as `trans.execute()[-1][0][0]`)

Since the return type of the function is different and this has been already criticized in another parts of the code, I would like to hear form others before doing this change.